src/main/java/net/earthcomputer/clientcommands/features/ClientCommandFunctions.java

```diff
@@ -0,0 +1,252 @@
+package net.earthcomputer.clientcommands.features;
+
+import com.google.common.collect.ImmutableList;
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.ParseResults;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.DynamicCommandExceptionType;
+import com.mojang.logging.LogUtils;
+import net.earthcomputer.clientcommands.TempRules;
+import net.earthcomputer.clientcommands.command.VarCommand;
+import net.fabricmc.fabric.api.client.command.v2.ClientCommandManager;
+import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ServerInfo;
+import net.minecraft.server.command.CommandManager;
+import net.minecraft.server.integrated.IntegratedServer;
+import net.minecraft.text.Text;
+import net.minecraft.util.FileNameUtil;
+import net.minecraft.util.WorldSavePath;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.IntConsumer;
+import java.util.stream.Stream;
+
+public class ClientCommandFunctions {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    private static final Path FUNCTION_DIR = FabricLoader.getInstance().getConfigDir().resolve("clientcommands").resolve("functions");
```

Comment:
I suppose this applies to all commands involving file storage, but should we use the static variable `configDir` from `ClientCommands.java`? If not we could make that variable a local variable.

Reply:
I forgot that existed